docs/spec/lightclient/detection/detection.md

```diff
@@ -375,14 +656,110 @@ func FetchLightBlock(peer PeerID, height Height) LightBlock
 ```
 See the [verification specification][verification] for details.
 
+
+#### **[LCD-FUNC-SUBMIT.1]:**
+```go
+func SubmitProofOfFork(pof LightNodeProofOfFork) Result
+```
+**TODO:** finalize what this should do, and what detail of
+  specification we need.
+- Implementation remark
+- Expected precondition
+    - none
+- Expected postcondition
+    - submit evidence to primary and the secondary in *pof*, that is,
+      to
+	     - `pof.PrimaryTrace[1].Provider`
+	     - `pof.SecondaryTrace[1].Provider`
+    - **QUESTION** minimize data? We could submit to the primary only
+      the trace of the secondary, and vice versa. Do we need to spell
+      that out here? (Also, by [LCD-INV-TRUSTED-AGREED.1], we do not
+      need to send `pof.TrustedBlock`)
+	- **FUTURE WORK:** we might send *pof* to primary or all
+      secondaries or broadcast to all full nodes. However, in evidence
+      detection this might need that a full node has to check a *pof*
+      where both traces are not theirs. This leads to more complicated
+      logic at the full node, which we do not need right now.
+
+- Error condition
+    - none
+
 ### Auxiliary Functions (Local)
 
+#### **[LCD-FUNC-CROSS-CHECK.1]:**
 
 ```go
-Replace_Secondary(addr Address)
+func CrossCheck(peer PeerID, testedLB LightBlock) (result) {
+	sh := FetchLightBlock(peer, testedLB.Height);
+		// as the check below only needs the header, it is sufficient
+		// to download the header rather than the LighBlock
+    if testedLB.Header == sh.Header {
+	    return OK
+	}
+	else {
+	    return DoesNotMatch
+	}
+}
 ```
+- Implementation remark
+    - download block and compare to previously downloaded one.
+- Expected precondition
+- Expected postcondition
+- Error condition
+
+
+#### **[LCD-FUNC-REPLACE-PRIMARY.1]:**
+```go
+Replace_Primary()
+```
+**TODO:** formalize conditions
+- Implementation remark
+    - the primary is replaced by a secondary, and lightblocks above
+      trusted blocks are removed
```

Comment:
Maybe those blocks just need to go through fork detection procedure instead of removing them.

Reply:
We could do that. It will just make the control flow a bit more complicated.